multiplexer/abci/multiplexer.go

```diff
@@ -450,9 +454,12 @@ func openTraceWriter(traceWriterFile string) (w io.WriteCloser, err error) {
 
 // getApp gets the appropriate app based on the latest application version.
 func (m *Multiplexer) getApp() (servertypes.ABCI, error) {
```

Comment:
it seems like everywhere getApp is called we have a context. did you try passing the context here for cancellation? it may be hard to verify on all flows as im not sure comet uses contexts correctly for shutdown?